blog/2021-11-04-introducing-pwa.md

```diff
@@ -0,0 +1,168 @@
+
+export function CacheableDashboard({ id }) {
+  const { startRecording } = useCacheableSection(id);
+
+  return (
+    <>
+      <Button onClick={() => startRecording()}>Make available offline</Button>
+      <CacheableSection id={id} loadingMask={<LoadingMask />}>
+        <Dashboard id={id} />
+      </CacheableSection>
+    </>
+  );
+}
+```
+
+When the “Make available offline” button is clicked and `startRecording` is called, the children of the `<CacheableSection>` component with the same `id` as the `useCacheableSection` hook will reload, and the provided loading mask will be rendered while the cacheable section records the requests that the section makes. Once the recording is done, that section can be accessed while offline!
+
+To learn more about the usage of cacheable sections, including an example of a Loading Mask, check out the [“Usage” section](https://runtime.dhis2.nu/#/advanced/offline?id=usage) in the App Runtime docs.
+
+### Online status
+
+If you took a look at the 2.37 version of the Dashboards app while offline, you may have noticed that many features of the app are disabled while offline, and some items in the interface change appearance.
+
+The App Runtime provides a `useOnlineStatus` hook that can access the online or offline status of the app, which the Dashboards app uses to enable or disable those features. Using the hook looks like this:
+
+```jsx
+import { useOnlineStatus } from "@dhis2/app-runtime";
+
+const OnlineComponent = () => {
+  const { online, offline } = useOfflineStatus();
+
+  return <p>{online ? "Online" : "Offline"}</p>;
+};
+```
+
+## An important caveat
+
+These features are exciting and the opportunity to take apps offline can be very useful when taken into areas with low network connectivity, but there is an important limitation that you should know before using the PWA features.
+
+In its current implementation, **cached data is not encrypted** when stored offline, and a malicious actor could inspect access and inspect the cached data if they gain access to a user’s device. For this reason, it’s important to **not use** PWA features in an app that will handle sensitive data, especially if the app is likely to be used on shared devices.
+
+Encryption of the offline data will be the next feature for these PWA tools however, so keep an eye out for the announcement of that release.
+
+## Want to know more?
+
+If you would like to learn more about adding offline caching, cacheable sections, or using other PWA features to your app, take a look at the documentation at the [App Platform](https://platform.dhis2.nu/#/pwa/pwa) and [App Runtime](https://runtime.dhis2.nu/#/advanced/offline) sites which go into greater detail about the API and provide more examples.
+
+In the near future we will also post an detailed article about the technical development of these PWA features because we’re using these tools in a pretty unique and cool way that we want to share, so keep an eye out for that post!
```

Comment:
```suggestion
In the near future we will also post a detailed article about the technical development of these PWA features because we’re using these tools in a pretty unique and cool way that we want to share, so keep an eye out for that post!
```